### PR TITLE
feat: unify OTEL tracing and ART

### PR DESCRIPTION
**Pull request by thisisnithin**
## Motivation and Context

image

## TODO

- [ ] Tests or benchmark included
- [ ] Documentation is changed or added on [https://app.gitbook.com/](https://app.gitbook.com/)
- [x] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)

### CODE REVIEW

**Comment by jensneuse** [comment]
Small feedback on aligning the texts.
image

**Review by jensneuse — APPROVED** [review]

File: router/core/graphql_prehandler.go [hunk 1]
```diff
@@ -114,6 +113,36 @@ func (h *PreHandler) Handler(next http.Handler) http.Handler {
 			otel.WgOperationProtocol.String(OperationProtocolHTTP.String()),
 		}
 
+		if h.enableRequestTracing {
```
Comment (StarpTech) [inline router/core/graphql_prehandler.go:116]: Did you test all the cases?

File: router/core/graphql_prehandler.go [hunk 2]
```diff
@@ -209,8 +250,12 @@ func (h *PreHandler) Handler(next http.Handler) http.Handler {
 
 		engineNormalizeSpan.End()
 
+		if !traceOptions.ExcludeNormalizeStats {
+			traceTimings.EndNormalize()
+		}
+
 		if h.traceExportVariables {
-			// At this stage the variables are normalized
+			// At this stage the variables are norfmt.Printf("test: %d", resolve.GetDurationNanoSinceTraceStart(tt.ctx))malized
```
Comment (StarpTech) [inline router/core/graphql_prehandler.go:258]: Remove?

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: router/pkg/trace/art.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
+package trace
+
+import (
```
Comment (StarpTech) [inline router/pkg/trace/art.go:3]: trace package only contains files about OTEL tracing. Could you put this in e.g ART package?

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: router/core/graphql_prehandler.go (hunk 1 above)
Comment (thisisnithin) [reply router/core/graphql_prehandler.go:116]: Yep

File: router/core/graphql_prehandler.go (hunk 2 above)
Comment (thisisnithin) [reply router/core/graphql_prehandler.go:258]: Done

File: router/pkg/trace/art.go (hunk 3 above)
Comment (thisisnithin) [reply router/pkg/trace/art.go]: Done

**Review by StarpTech — APPROVED** [review]
LGTM

**Pull request merged**